functions/controllers/application/index.js

```diff
@@ -0,0 +1,120 @@
+const functions = require("firebase-functions");
+const express = require("express");
+const cors = require("cors");
+const formidable = require("formidable-serverless");
+
+const { jwtCheck, hasUpdateApp, hasReadApp } = require("../../utils/middleware");
+const {
+  createAppObject,
+  validateAppData,
+  validateResume,
+  isValidFileData,
+  getNewFileName,
+} = require("../../utils/application");
+const { queryDocument, setDocument, uploadFile } = require("../../utils/database");
+
+const application = express();
+application.disable("x-powered-by");
+application.use(express.json());
+
+const auth0Config = functions.config().auth;
+const corsConfig = auth0Config ? auth0Config.cors : "";
+
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+application.use(cors(corsOptions));
+
+/* TODO: 
+  Update createAppObject and validateAppObject with schema
+  Update getNewFileName to get unique filenames
+  Unit Test Functions
+*/
+application.post("/submit", jwtCheck, hasUpdateApp, async (req, res) => {
+  try {
+    const form = new formidable.IncomingForm();
+    return await form.parse(req, async (err, fields, files) => {
+      if (err) {
+        functions.logger.log("Form Error: " + err);
```

Comment:
I was getting an error here, so I wasn't able to test this endpoint.

```
 {"severity":"INFO","message":"Form Error: TypeError: this._parser.write is not a function"}
```

I might've installed the wrong version of formidable?